torch/_higher_order_ops/invoke_subgraph.py

```diff
@@ -420,18 +420,41 @@ def backward(
                 filtered_grad_outs.append(o)
         filtered_grad_outs = tuple(filtered_grad_outs)
 
+        # Important note - Even though the forward graph can be same for
+        # different invoke_subgraphs, the backward graph can be different
+        # because the tangent strides can be different. So, here we cache on
+        # tangent_metadata in addition to identifier
+        from torch._guards import detect_fake_mode
+        from torch._subclasses._fake_tensor_utils import _CacheKeyState
+        from torch._subclasses.fake_tensor import extract_tensor_metadata
+
+        fake_mode = detect_fake_mode(primals + filtered_grad_outs)
+        state = _CacheKeyState(fake_mode.shape_env)
+
+        tangent_metadata: list[object] = []
+        for tangent in filtered_grad_outs:
+            metadata = extract_tensor_metadata(tangent)
+            metadata._flatten_into(tangent_metadata, fake_mode, state)
```

Comment:
I never had a chance to review the initial FakeTensorMode caching for symbolic shapes :( but from some quick reading this looks reasonable to me